### PR TITLE
Christian input

### PR DESCRIPTION
**Pull request by altffyra**
### **New feature Input:**

  label: string - Label text for input
  variant?: string - variants for input 'text', 'number', 'email', 'password'
  validationText?: string - text to inform of success or faliure, activated when isValid is true or false
  isValid?: boolean | null - Boolean to activate green or red colors and icon to inform user of state.
  placeholder?: string | undefined - Placeholder text in input.
  disabled?: boolean - Boolean to deactivate controls and dim visibility
  modelValue?: string | null - v-model value
  helpertext?: string - Text to guide the user
  size?: string - set size of the input and font size.
  
 ### Usage:
 To use the component you need to v-model it to a value then add what props you want ie:
``<Inputvue v-model='inputRef' text='Label' helpertext='Please input value' size='sm'>``
will result in this: 
image

for further usage consult Storybook documentation. 

### Working:
5 Vitests
Storybook
Fully working module

### Not working: 
Email validation

### CODE REVIEW

File: src/components/input/input.vue [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        <div v-if="accordianRef" class="iconWrapper">
+          <AsyncIcon
+            v-if="accordianRef"
+            class="dropdown h2"
+            :class="[accordianSwitch ? 'toggledAccordian' : '']"
+            icon="caret-down"
+            @click="toggleAccordian"
+          />
+        </div>
+      </h1>
```
Comment (adamfagerstrom) [inline src/components/input/input.vue:122]: I left a comment in the number input that correlates to this one aswell. Use label and then name in the input to connect them!

File: src/components/input/input.vue [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+} from 'vue';
+import {
+  inputSize,
+  defaultVariantMixin,
+  inputVariant,
+} from '@/helpers/mixins/jsMixins';
+import('@/components/button/button.vue');
+
+type InputProps = {
+  text: string
```
Comment (adamfagerstrom) [inline src/components/input/input.vue:19]: Maybe label as a propname?

File: src/components/input/input.vue [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  () => showPassword.value,
+  (newVal) => {
+    newVal
+      ? (variantMiddleware.value = 'text')
+      : (variantMiddleware.value = 'password');
+  }
+);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function emitInput(e: any) {
```
Comment (adamfagerstrom) [inline src/components/input/input.vue:75]: Also same as number-input comments :)

**Review by adamfagerstrom — COMMENTED** [review]
Very nice Christian, you have some small fixes that you can "steal" from the number-input regarding fonts, a11y e.tc. But same here. Great work and a great start for a input component!

File: src/components/input/input.scss [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    user-select: none;
+    border-color: map-get(colors.$input-colors, danger);
+    svg {
+      color: map-get(colors.$input-colors, danger);
+    }
+  }
+}
+
+.userInstructions {
+  user-select: none;
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/input/input.scss]: removed!

File: src/components/input/input.vue [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  size: 'md',
+});
+
+const accordianRef: Ref = ref<boolean>(false);
+slots.sentContent ? (accordianRef.value = true) : (accordianRef.value = false);
+const size: Ref = ref<string>(props.size);
+const variantMiddleware: Ref = ref<string>(props.variant);
+const showPassword: Ref = ref<boolean>(false);
+const type: Ref = ref<string>(props.variant);
+const accordianSwitch: Ref = ref<boolean>(false);
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/input/input.vue]: removed!

File: src/components/input/input.vue (hunk 1 above)
Comment (adamfagerstrom) [reply src/components/input/input.vue]: Good with label, same as with-number input about the font-sizes here!

File: src/components/input/input.vue [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+</script>
+
+<template>
+  <div class="input-body">
+    <div :class="[size, 'input-body__top-wrapper', props.disabled ? 'disabled' : '']">
+      <label :for="labelRef" class="body1">
+        {{ props.label }}
+        <div v-if="accordianRef" class="icon-wrapper">
+          <AsyncIcon
+            v-if="accordianRef"
```
[the remark this answers is not in the document]
Comment (saradalsman) [reply src/components/input/input.vue:123]: Also, add tabindex so you can access it via keyboard. And `cursor: pointer` style

File: src/components/input/input.vue [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        :class="[
+          props.disabled ? 'disabled' : '',
+          props.isValid == true ? 'success-message-text' : '',
+          props.isValid == false ? 'error-message-text' : '',
+          'body2'
+        ]"
+      >
+        {{ props.validationText }}
+      </p>
+      <p class="helper-message-text body2">
```
Comment (saradalsman) [inline src/components/input/input.vue:210]: Add v-if="props.helpertext" so the p-element only renders if it has content

Comment (altffyra) [reply src/components/input/input.vue:210]: v-if="props.helpertext != ''" added

File: src/components/input/input.vue [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+      >
+        <AsyncIcon
+          v-if="showPassword == false"
+          icon="eye"
+          @click="changePasswordVisibility"
+        />
+        <AsyncIcon
+          v-if="showPassword == true"
+          icon="eye-slash"
+          @click="changePasswordVisibility"
```
Comment (saradalsman) [inline src/components/input/input.vue:194]: Would be nice with tabindex and `cursor: pointer` on the eye icon as well

Comment (altffyra) [reply src/components/input/input.vue:194]: already added in new version

Comment (altffyra) [reply src/components/input/input.vue:194]: currently thinking about autofocus to next icon becouse it loses focus when using eye

File: src/components/input/input.stories.ts [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    <div style="display: flex; flex-direction: column;">
+    <foam-input  variant="text" helpertext="helpertext" validationText="validationText" :isValid=false v-bind="args"  />
+    <foam-input  variant="text" 
+    helpertext="helpertext"  validationText="validationText"  :isValid=true v-bind="args"  />
+    </div>
+  `,
+  };
+};
+
+export const Accordian: StoryFn<typeof FoamInput> = () => {
```
Comment (saradalsman) [inline src/components/input/input.stories.ts:196]: It should be spelled "Accordion" instead of "Accordian" 🙃 so change this everywhere

Comment (saradalsman) [reply src/components/input/input.stories.ts:196]: There are a couple left, in input.md and input.stories.ts

**Review by adamfagerstrom — APPROVED** [review]
After the typo, this should definatley be ready for merge! Great work!

**Review by saradalsman — APPROVED** [review]
Great job! 🙌

**Pull request merged**